### PR TITLE
Clickable transport and stops + refactor stop modal

### PR DESCRIPTION
**Pull request by hiba9201**
- updated redux actions for not clearing prev state
- refactored Stop modal
- moved location marker to bottom
- made clickable transport and stop in card for stop and transport respectively

### CODE REVIEW

File: client/components/Map/Stops/Sidebar/StopsList/Item/MapStopsSidebarStopsListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        <div
+            key={`${vehicle.route}-${vehicle.type}-${vehicle.arriveTime}`}
+            className={cn(styles.MapStopsSidebarVehicle, {
+                [styles.MapStopsSidebarVehicle_isSelected]: currentVehicleRoute === vehicle.route,
+            })}
+            onClick={setSelectedVehicle}
+            style={{
+                // @ts-ignore
+                '--vehicle-color': VEHICLE_TYPE_COLORS[vehicle.type],
+            }}
```
Comment (a-chabin) [inline client/components/Map/Stops/Sidebar/StopsList/Item/MapStopsSidebarStopsListItem.tsx:69]: Let's remove `@ts-ignore`

```suggestion
            style={{ "--vehicle-color":  VEHICLE_TYPE_COLORS[type] } as React.CSSProperties}
```

Comment (hiba9201) [reply client/components/Map/Stops/Sidebar/StopsList/Item/MapStopsSidebarStopsListItem.tsx:69]: yeah, thanks, forgot to fix it!

File: common/types/strapi.ts [hunk 2]
```diff
@@ -6,20 +6,16 @@ export enum StrapiContentTypes {
     UnitInfo = 'unit-info',
 }
 
-export interface StrapiStop {
+export interface StrapiBase<A> {
```
Comment (a-chabin) [inline common/types/strapi.ts:9]: Wht not `<T>`?

Comment (hiba9201) [reply common/types/strapi.ts:9]: A – Attributes?

File: client/components/Map/Vehicles/Sidebar/MapVehiclesSidebar.tsx [hunk 3]
```diff
@@ -187,8 +195,40 @@ export function MapVehiclesSidebar({
...
+    );
+
     return (
-        <div className={cn(styles.MapVehiclesSidebar)}>
+        <div
+            className={cn(styles.MapVehiclesSidebar)}
+            style={{
+                // @ts-ignore
+                '--vehicle-color': VEHICLE_TYPE_COLORS[type],
+            }}
```
Comment (a-chabin) [inline client/components/Map/Vehicles/Sidebar/MapVehiclesSidebar.tsx:230]: Let's remove `// @ts-ignore`

```suggestion
            style={{ '--vehicle-color': VEHICLE_TYPE_COLORS[type] } as React.CSSProperties}
```

File: common/types/strapi.ts (hunk 2 above)
Comment (hiba9201) [reply common/types/strapi.ts:9]: I can rename it to `AttributesType` for example?

File: client/components/Map/Stops/Sidebar/StopsList/MapStopsSidebarStopsList.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        <MapStopsSidebarRow mix={styles.MapStopsSidebarStopsList}>
+            {stopInfo.map((vehicle) => (
+                <MapStopsSidebarStopsListItem
+                    vehicle={vehicle}
+                    key={`${vehicle.route}-${vehicle.arriveTime}`}
+                />
+            ))}
+            {stopInfo.length === 0 && (
+                <Typography variant="h4">
+                    {t('О нет! В ближайшее время транспорта тут не будет')}
```
Comment (a-chabin) [inline client/components/Map/Stops/Sidebar/StopsList/MapStopsSidebarStopsList.tsx:26]:
```suggestion
                    {t('О, нет! В ближайшее время транспорта тут не будет')}
```

File: client/components/Map/Stops/Sidebar/StopsList/Item/MapStopsSidebarStopsListItem.tsx (hunk 1 above)
Comment (a-chabin) [reply client/components/Map/Stops/Sidebar/StopsList/Item/MapStopsSidebarStopsListItem.tsx:69]: yep

it will be more readable!

**Review by a-chabin — APPROVED** [review]
LGTM!

**Pull request merged**